src/table.ts

```diff
@@ -745,22 +745,35 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
       filter = Filter.parse(options.filter);
     }
 
-    const userStream = new PassThrough({objectMode: true});
-    const end = userStream.end.bind(userStream);
-    userStream.end = () => {
+    let chunkTransformer: ChunkTransformer;
+    let rowStream: Duplex;
+
+    let userCanceled = false;
+    const userStream = new PassThrough({
+      objectMode: true,
+      transform(row, _encoding, callback) {
+        if (userCanceled) {
+          callback();
+          return;
+        }
+        callback(null, row);
+      },
+    });
+    const originalEnd = userStream.end.bind(userStream);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    userStream.end = (chunk?: any, encoding?: any, cb?: () => void) => {
       rowStream?.unpipe(userStream);
+      rowStream.removeListener('end', originalEnd);
```

Comment:
can you add 2 helper functions rowStreamPipe & rowStreamUnpipe  and add comments that we need to do special handling of the end event to re-direct the handler to the originalEnd